packages/xml-views-validation/src/utils/custom-class.ts

```diff
@@ -0,0 +1,36 @@
+import { SVG_NS, TEMPLATING_NS, XHTML_NS } from "./special-namespaces";
+import { XMLElement } from "@xml-tools/ast";
+import { resolveXMLNS } from "@ui5-language-assistant/logic-utils";
+
+// Heuristic to limit false positives by only checking tags starting with upper
+// case names, This would **mostly** limit the checks for things that can actually be
+// UI5 Elements / Controls.
+export function isCustomClass(xmlElement: XMLElement): boolean {
+  // This will never happen - it was checked before calling this function
+  /* istanbul ignore if */
+  if (xmlElement.name === null) {
+    return false;
+  }
+
+  return /^[A-Z]/.test(xmlElement.name) && !isNoneUI5id(xmlElement);
```

Comment:
I am aware this is copy/paste from my previous code.
But perhaps take the chance to improve the name.

The double negative of `!isNoneUI5id` can be confusing.
But I am having trouble thinking of another name :(